Support the `in` operator with str

ansley opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47057 Support the `in` operator with str**

Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

Commit:
Support the `in` operator with str

[ghstack-poisoned]

Commit:
Update on "Support the `in` operator with str"




[ghstack-poisoned]

eellison commented on torch/csrc/jit/runtime/register_prim_ops.cpp:
```diff
@@ -818,6 +818,17 @@ RegisterOperators reg(
              "aten::__contains__.str_list(str[] l, str item) -> bool"),
          listContains<std::string>,
          aliasAnalysisFromSchema()),
+     OperatorGenerator(
```
Another way to support `__contains__` would be to use our mechanism to register operators that desguar to other operators in `builtin_functions.cpp`.  I think you would just add to the ops section here https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/frontend/builtin_functions.cpp#L65

We haven't been as judicious as we could have been in composing builtins in terms of other builtins, but I think it's generally a good idea to have less operators because they contribute to mobile size, and having fewer operators would make it easier to do things like do codegen for builtin python types down the road.

eellison commented on test/test_jit.py:
```diff
@@ -6735,6 +6735,11 @@ def complicated_arithmetic_operation():
+        def fn() -> bool:
+            return "a" in "abcd"
+        self.checkScript(fn, ())
+
```
Maybe add a test that uses `__contains__` as a method (granted it is an uncommon way of calling it)

eellison reviewed (commented):
Looks good, one comment abt using `builtin_functions.cpp`

Commit:
Update on "Support the `in` operator with str"




[ghstack-poisoned]

eellison reviewed (approved):
lgtm!

Commit:
Update on "Support the `in` operator with str"


Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

[ghstack-poisoned]

Commit:
Update on "Support the `in` operator with str"


Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

[ghstack-poisoned]

Commit:
Update on "Support the `in` operator with str"


Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

[ghstack-poisoned]

Commit:
Update on "Support the `in` operator with str"


Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

[ghstack-poisoned]

Commit:
Update on "Support the `in` operator with str"


Differential Revision: [D24863370](https://our.internmc.facebook.com/intern/diff/D24863370)

[ghstack-poisoned]
